nni/algorithms/compression/v2/pytorch/utils/constructor_helper.py

```diff
@@ -0,0 +1,126 @@
+            param_ids = [id(p) for p in params]
+            param_group['params'] = [name for name, p in model.named_parameters() if id(p) in param_ids]
+
+        return param_groups
+
+    def names2params(self, wrapped_model: Module, origin2wrapped_name_map: Dict, params: List[Dict]) -> List[Dict]:
+        param_groups = deepcopy(params)
+        for param_group in param_groups:
+            wrapped_names = [origin2wrapped_name_map.get(name, name) for name in param_group['params']]
+            param_group['params'] = [p for name, p in wrapped_model.named_parameters() if name in wrapped_names]
+        return param_groups
+
+    def call(self, wrapped_model: Module, origin2wrapped_name_map: Dict) -> Optimizer:
+        args = deepcopy(self.args)
+        kwargs = deepcopy(self.kwargs)
+
+        if 'params' in kwargs:
+            kwargs['params'] = self.names2params(wrapped_model, origin2wrapped_name_map, kwargs['params'])
+        else:
+            args[0] = self.names2params(wrapped_model, origin2wrapped_name_map, args[0])
+
+        return self.callable_obj(*args, **kwargs)
+
+    @staticmethod
+    def from_trace(model: Module, optimizer_trace: SerializableObject):
+        assert isinstance(optimizer_trace, SerializableObject), \
+            'Please use nni.algorithms.compression.v2.pytorch.utils.trace to wrap the optimizer class before initialize the optimizer.'
+        assert isinstance(optimizer_trace, Optimizer), \
+            'It is not an instance of torch.nn.Optimizer.'
+        return OptimizerConstructHelper(model,
+                                        optimizer_trace._get_nni_attr('symbol'),
+                                        *optimizer_trace._get_nni_attr('args'),
+                                        **optimizer_trace._get_nni_attr('kwargs'))
+
+
+class LRSchedulerConstructHelper(ConstructHelper):
+    def __init__(self, lr_scheduler_class: Callable, *args, **kwargs):
+        args = list(args)
+        if 'optimizer' in kwargs:
+            kwargs['optimizer'] = None
+        else:
+            args[0] = None
+        super().__init__(lr_scheduler_class, *args, **kwargs)
+
+    def call(self, optimizer: Optimizer) -> _LRScheduler:
+        args = deepcopy(self.args)
+        kwargs = deepcopy(self.kwargs)
+
+        if 'optimizer' in kwargs:
+            kwargs['optimizer'] = optimizer
+        else:
+            args[0] = optimizer
+
+        return self.callable_obj(*args, **kwargs)
+
+    @staticmethod
+    def from_trace(lr_scheduler_trace: SerializableObject):
+        assert isinstance(lr_scheduler_trace, SerializableObject), \
+            'Please use nni.algorithms.compression.v2.pytorch.utils.trace to wrap the lr scheduler class before initialize the scheduler.'
+        assert isinstance(lr_scheduler_trace, _LRScheduler), \
+            'It is not an instance of torch.nn.lr_scheduler._LRScheduler.'
+        return LRSchedulerConstructHelper(lr_scheduler_trace._get_nni_attr('symbol'),
```

Comment:
You can use `.trace_symbol`, `.trace_args` and `.trace_kwargs`.

Reply:
fixed